attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/service/AttributeMaskingServiceErrorTest.java

```diff
@@ -111,11 +112,16 @@ void testMaskingFailure() {
         // When masking
         final AuditableAttributeMaskingResponse subject = this.attributeMaskingService.maskResourceAttributes(attributeMaskingRequest);
         // Then the service suppresses exception and populates Audit object
-        assertThat(subject.getAuditErrorMessage().getError().getMessage())
+        assertThat(subject)
                 .as("verify that exception is propagated into an auditable object and returned")
+                .extracting(AuditableAttributeMaskingResponse::getAuditErrorMessage)
+                .isNotNull()
+                .extracting(AuditErrorMessage::getError)
+                .extracting(Throwable::getMessage)
                 .isEqualTo("Cannot mask");
 
-        assertThat(subject.getAttributeMaskingResponse())
+        assertThat(subject)
+                .extracting(AuditableAttributeMaskingResponse::getAttributeMaskingResponse)
                 .as("verify that auditable object has no payload")
                 .isNull();
```

Comment:
Might be worth moving the `.as()` so it's in the same place in the tests. More of an aesthetic change though.
```suggestion
        assertThat(subject)
                .as("verify that auditable object has no payload")
                .extracting(AuditableAttributeMaskingResponse::getAttributeMaskingResponse)
                .isNull();
```